pkg/nfd-master/nfd-master.go

```diff
@@ -605,20 +604,81 @@ func (m *nfdMaster) nfdAPIUpdateOneNode(nodeName string) error {
 	return nil
 }
 
-func (m *nfdMaster) refreshNodeFeatures(cli *kubernetes.Clientset, nodeName string, annotations, labels map[string]string, features *nfdv1alpha1.Features) error {
+// validateExtendedResources validates extended resources and returns a map
+// of valid extended resources.
+func (m *nfdMaster) validateExtendedResources(features *nfdv1alpha1.Features, extendedResources ExtendedResources) ExtendedResources {
+	outExtendedResources := ExtendedResources{}
+	deniedNs := map[string]struct{}{"kubernetes.io": {}}
+	deniedWildCarNs := map[string]struct{}{".kubernetes.io": {}}
+	allowedNS := map[string]struct{}{nfdv1alpha1.ExtendedResourceNs: {},
+		nfdv1alpha1.FeatureLabelNs: {}}
+	for extendedResource, capacity := range extendedResources {
+		// Add possibly missing default ns
+		extendedResource = addNs(extendedResource, nfdv1alpha1.ExtendedResourceNs)
+		// Check if given NS is allowed
+		ns, _ := splitNs(extendedResource)
+		if isNamespaceDenied(ns, deniedWildCarNs, deniedNs) {
+			if _, ok := allowedNS[ns]; !ok {
+				klog.Errorf("Namespace %q is not allowed. Ignoring Extended Resource  %q\n", ns, extendedResource)
+				continue
+			}
+		}
+
+		// Dynamic Value
+		if strings.HasPrefix(capacity, "@") {
+			// Remove the @ prefix
+			capacity = strings.TrimPrefix(capacity, "@")
+			// capacity is a string in the form of attribute.featureset.elements
+			split := strings.SplitN(capacity, ".", 3)
+			if attrFeatureSet, ok := features.Attributes[split[0]+"."+split[1]]; ok {
+				if element, ok := attrFeatureSet.Elements[split[2]]; ok {
+					q, err := k8sQuantity.ParseQuantity(element)
+					if err != nil {
+						klog.Errorf("bad label value %s encountered for extended resource: %s", q.String(), extendedResource, err.Error())
+						continue
+					}
+					outExtendedResources[extendedResource] = q.String()
+					continue
+				}
+				klog.Errorf("element %s not foundon feature %s. Ignoring Extended Resource %q/n", split[2], split[0]+"."+split[1], extendedResource)
```

Comment:
remove `/n`
```suggestion
				klog.Errorf("element %s not foundon feature %s. Ignoring Extended Resource %q", split[2], split[0]+"."+split[1], extendedResource)
```

Reply:
After taking https://github.com/kubernetes-sigs/node-feature-discovery/pull/1099#discussion_r1157614130 , this comment is invalid